docs/riff_system_install.md

```diff
@@ -10,6 +10,19 @@ If an 'istio-system' namespace isn't found then the it will be created and Istio
 
 Use the '--node-port' flag when installing on Minikube and other clusters that don't support an external load balancer.'
 
+Use the '--manifest' flag to specify the path of a manifest file which provides the URLs of the YAML files to be applied.
+The manifest file contents should be of the following form:
+
+version: 0.1
```

Comment:
the format here is different than what I see in the `cmd/commands/system.go` file.

I kind of like the grouping here, so if we could have the following:
```
istio: 
- https://path/to/istio.yaml
knative:
- https://path/to/knative-build.yaml
- https://path/to/knative-serving.yaml
- https://path/to/knative-eventing.yaml
- https://path/to/stub-clusterbus.yaml
namespace:
- https://path/to/riff-buildtemplate.yaml
```
then we would have more flexibility in including one, two ore more files for each group

Reply:
I was considering a grouping along those lines, especially since @scothis suggested it in the issue. However, there is some special processing which doesn't lend itself to this approach.

For istio, there is a 5 second delay between installing the CRDs and installing the other istio resources. Apart from the obvious race involved when deploying to a very sluggish system, I was trying not to break the current behaviour. I guess I could introduce a 5 second delay between all the items in the istio group, but that feels like a bit of a hack.

Similarly, there is a change applied to the serving YAML if and only if `--node-port` is specified. I am a bit nervous about applying this change to all the YAML files in the `knative` group because we can't tell what YAML files users will add to each group and whether such a global change would be correct or not.

Reply:
@trisberg As per our chat, I'll rework in terms of an istio array with 5 second intervals between each pair and a knative array with a more specific global change. Thanks for helping me work that through. /cc @scothis

Reply:
The current Istio config is a hack to work around a bug that has been fixed. There are still potential issues when applying the new istio yaml, but that can be solved by a retry on failure. A retry is not a bad idea for any applied yaml.

The NodePort/LoadBalancer swap should be "relatively" safe, we can make the match more selective if there are false positives.

Reply:
how do we know when to retry? we don't really know what's in the yaml files, orwhat namespaces they target except for the istio install

Reply:
> how do we know when to retry?

When the command you're running exits with a non-zero code.

Reply:
The 5 second delay between istio YAMLs may be necessary if someone uses a manifest to install istio earlier than v1.0.1. I'd prefer to defer retries to a separate issue to avoid scope creep in this issue.

Reply:
I'll open a separate issue for retry